### PR TITLE
Clear out unused compiler options when transpiling

### LINKED ISSUE #9173: transpileModule() fails when lib option was given
**TypeScript Version:** 

Version 1.9.0-dev.20160614-1.0

**Code**

```js
const typescript = require("typescript");
const input = `export class Test { }`;
const output = typescript.transpileModule(input,
    {
        compilerOptions: {
            target: "es5",
            lib: ["es2015", "dom"],
            module: "commonjs",
        },
        reportDiagnostics: true
    }
);
console.log(output.diagnostics[0].messageText);
```

**Expected behavior:**
Compiles without error

**Actual behavior:** 
```
Option 'lib' cannot be specified with option 'noLib'.
```

Since transpileModule() forces ```noLib``` flag i'd expect it doesn't take account into ```lib``` option

### PR DESCRIPTION
**Pull request by mhegazy**
Take-2 Fixes #9173.

Clear out more settings, and add tests for all the settings that were not covered before.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -2001,9 +2001,17 @@ namespace ts {
         // so pass --noLib to avoid reporting a file not found error.
         options.noLib = true;
 
-        // Clear out the lib and types option as well
+        // Clear out other settings that would not be participate in transpiling this module
```
Comment (vladima) [inline src/services/services.ts]: nit: would not be used

Comment (mhegazy) [inline src/services/services.ts]: done

**Comment by yuit** [comment]
@mhegazy  you will want to merge master in sicne we have change our unittests/transpile to use baselines instead

**Comment by mhegazy** [comment]
@yuit can you take a look?

**Comment by yuit** [comment]
@mhegazy sorry didn't realize my comment on Friday didn't show up. The change looks good. though I think we should pull transpile option test out as the file is growing

**Pull request merged**